From c28a16dbd06cfb44d7030cc3fadd4bd2c54961a9 Mon Sep 17 00:00:00 2001
From: Zheng Yang <zhengyang@rock-chips.com>
Date: Thu, 7 Dec 2017 09:40:06 +0800
Subject: [PATCH] drm/rockchip: inno_hdmi: fix audio infoframe

According to HDMI CTS 7-31, audio sample width and frequency
should be zero.

Change-Id: Ida37483e3f58e152e6a1c55d8bb81d0e9e0fb2ed
Signed-off-by: Zheng Yang <zhengyang@rock-chips.com>
---
 drivers/gpu/drm/rockchip/inno_hdmi.c | 36 ----------------------------
 1 file changed, 36 deletions(-)

diff --git a/drivers/gpu/drm/rockchip/inno_hdmi.c b/drivers/gpu/drm/rockchip/inno_hdmi.c
index 0746594448ea..d0724f776ad2 100644
--- a/drivers/gpu/drm/rockchip/inno_hdmi.c
+++ b/drivers/gpu/drm/rockchip/inno_hdmi.c
@@ -332,42 +332,6 @@ static int inno_hdmi_config_audio_aai(struct inno_hdmi *hdmi,
 
 	faudio->channels = audio->channels;
 
-	switch (audio->sample_width) {
-	case 16:
-		faudio->sample_size = HDMI_AUDIO_SAMPLE_SIZE_16;
-		break;
-	case 20:
-		faudio->sample_size = HDMI_AUDIO_SAMPLE_SIZE_20;
-		break;
-	case 24:
-		faudio->sample_size = HDMI_AUDIO_SAMPLE_SIZE_24;
-		break;
-	}
-
-	switch (audio->sample_rate) {
-	case 32000:
-		faudio->sample_frequency = HDMI_AUDIO_SAMPLE_FREQUENCY_32000;
-		break;
-	case 44100:
-		faudio->sample_frequency = HDMI_AUDIO_SAMPLE_FREQUENCY_44100;
-		break;
-	case 48000:
-		faudio->sample_frequency = HDMI_AUDIO_SAMPLE_FREQUENCY_48000;
-		break;
-	case 88200:
-		faudio->sample_frequency = HDMI_AUDIO_SAMPLE_FREQUENCY_88200;
-		break;
-	case 96000:
-		faudio->sample_frequency = HDMI_AUDIO_SAMPLE_FREQUENCY_96000;
-		break;
-	case 176400:
-		faudio->sample_frequency = HDMI_AUDIO_SAMPLE_FREQUENCY_176400;
-		break;
-	case 192000:
-		faudio->sample_frequency = HDMI_AUDIO_SAMPLE_FREQUENCY_192000;
-		break;
-	}
-
 	return inno_hdmi_upload_frame(hdmi, rc, &frame, INFOFRAME_AAI, 0, 0, 0);
 }
 
-- 
2.35.3

